presto-benchmark/src/main/java/com/facebook/presto/benchmark/PageProcessorBenchmark.java

```diff
@@ -0,0 +1,174 @@
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.presto.metadata.FunctionKind.SCALAR;
+import static com.facebook.presto.metadata.MetadataManager.createTestMetadataManager;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.sql.relational.Expressions.call;
+import static com.facebook.presto.sql.relational.Expressions.constant;
+import static com.facebook.presto.sql.relational.Expressions.field;
+import static java.util.stream.Collectors.toList;
+
+@State(Scope.Thread)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Fork(1)
+@Warmup(iterations = 3)
+@Measurement(iterations = 5)
+@BenchmarkMode(Mode.AverageTime)
+public class PageProcessorBenchmark
+{
+    private static final Metadata METADATA = createTestMetadataManager();
+    private static final Session TEST_SESSION = TestingSession.testSessionBuilder().build();
+
+    private PageProcessor processor;
+    private List<Page> inputPages;
+
+    private List<? extends Type> types;
+
+    @Param({ "10", "100", "1000", "2000" })
+    int pageCount;
+
+    @Param({ "1000", "2000", "4000" })
+    int positions;
+
+    @Setup
+    public void setup()
+    {
+        List<RowExpression> projections = getProjections();
+        types = projections.stream().map(RowExpression::getType).collect(toList());
+
+        inputPages = createPages(pageCount);
+        processor = new ExpressionCompiler(createTestMetadataManager()).compilePageProcessor(getFilter(), projections);
+    }
+
+    @Benchmark
+    public void rowOriented(Blackhole b)
+    {
+        for (int i = 0; i < pageCount; i++) {
+            PageBuilder pageBuilder = createPageBuilder();
+            b.consume(processor.process(null, inputPages.get(i), 0, inputPages.get(i).getPositionCount(), pageBuilder));
+        }
+    }
+
+    @Benchmark
+    public void columnOriented(Blackhole b)
+    {
+        for (int i = 0; i < pageCount; i++) {
+            PageBuilder pageBuilder = createPageBuilder();
+            b.consume(processor.processColumnar(null, inputPages.get(i), types));
+        }
+    }
+
+    public static void main(String[] args)
+            throws RunnerException
+    {
+        Options options = new OptionsBuilder()
+                .verbosity(VerboseMode.NORMAL)
+                .include(".*" + PageProcessorBenchmark.class.getSimpleName() + ".*")
+                .build();
+
+        new Runner(options).run();
+    }
+
+    private static RowExpression getFilter()
+    {
+        return new ConstantExpression(true, BooleanType.BOOLEAN);
+    }
+
+    private static List<RowExpression> getProjections()
+    {
+        return ImmutableList.of(
+                call(
+                        new Signature(FunctionRegistry.mangleOperatorName("ADD"), SCALAR, StandardTypes.BIGINT, StandardTypes.BIGINT, StandardTypes.BIGINT),
```

Comment:
Use the helpers in `com.facebook.presto.sql.relational.Signatures` (e.g., arithmeticExpressionSignature). The "mangleOperatorName" call is an internal detail and we should have other code depending on it.